src/util/system.cpp

```diff
@@ -979,6 +1028,217 @@ std::string ArgsManager::GetChainName() const
     return CBaseChainParams::MAIN;
 }
 
+namespace {
+
+    // Like std::getline, but includes the EOL character in the result
+    bool getline_with_eol(std::istream& stream, std::string& result)
+    {
+        int current_char;
+        current_char = stream.get();
+        if (current_char == std::char_traits<char>::eof()) {
+            return false;
+        }
+        result.clear();
+        result.push_back(char(current_char));
```

Comment:
Nit: I believe `(char)current_char` is the preferred form for casting the fundamentals types in the project :-)

Reply:
That's upstream: #11082